src/UI/Implementation/Crawler/EntriesYamlParser.php

```diff
@@ -0,0 +1,306 @@
+					" in file: ".$this->file_path." line ".$current_entry);
+		}else if($parser_state === self::PARSER_STATE_ENTRY){
+			throw $this->ef->exception(Exception\CrawlerException::ENTRY_WITH_NO_YAML_DESCRIPTION,
+					" in file: ".$this->file_path);
+		}else if($parser_state === self::PARSER_STATE_SEEKING_FUNCTION_NAME){;
+			throw $this->ef->exception(Exception\CrawlerException::ENTRY_WITHOUT_FUNCTION,
+					" in file: ".$this->file_path);
+		}
+		return $yaml_entries;
+	}
+
+	/**
+	 * @param	string $line
+	 * @return	string
+	 */
+	protected function purifyYamlLine($line){
+		return str_replace("* ", "", ltrim($line)).PHP_EOL;
+	}
+
+	/**
+	 * @param	string $yaml_entries
+	 * @return	array
+	 * @throws	Exception\CrawlerException
+	 */
+	protected function getPHPArrayFromYamlArray($yaml_entries){
+		$entries = array();
+		$parser = new Yaml\Parser();
+
+		foreach($yaml_entries as $yaml_entry){
+			try{
+				$entries[] = $parser->parse($yaml_entry);
+			}catch(\Exception $e){
+
+				throw $this->ef->exception(Exception\CrawlerException::PARSING_YAML_ENTRY_FAILED," file: ".$this->file_path."; ".$e);
+			}
+		}
+
+
+		array_walk_recursive($entries, function(&$item){
+			$item = rtrim($item,PHP_EOL);
+		});
+
+		return $entries;
+	}
+
+	/**
+	 * @param	array $entries_array
+	 * @return	Entry\ComponentEntries
+	 */
+	protected function getEntriesFromArray(array $entries_array){
+		$entries = new Entry\ComponentEntries();
+
+		foreach($entries_array as $entry_data){
+			$entries->addEntry($this->getEntryFromData($entry_data));
+		}
+
+		return $entries;
+	}
+
+	/**
+	 * @param	array $entry_data
+	 * @return	Entry\ComponentEntry
+	 * @throws	Exception\CrawlerException
+	 */
+	protected function getEntryFromData(array $entry_data){
+
+		$entry_data['title'] = self::fromCamelCaseToWords($entry_data['function_name']);
+
+		if(!array_key_exists("title",$entry_data) || !$entry_data['title'] || $entry_data['title'] ==""){
+			throw $this->ef->exception(Exception\CrawlerException::ENTRY_TITLE_MISSING," File: ".$this->file_path);
+		}
+		if(!array_key_exists("namespace",$entry_data) || !$entry_data['namespace'] || $entry_data['namespace'] ==""){
+			throw $this->ef->exception(Exception\CrawlerException::ENTRY_WITH_NO_VALID_RETURN_STATEMENT," File: ".$this->file_path);
+		}
+
+		$entry_data['id'] = str_replace("\\","",
+						str_replace("\\ILIAS\\UI\\","", str_replace("\\ILIAS\\UI\\Component\\","",$entry_data['namespace'])) )
+				.self::toUpperCamelCase($entry_data['title'], ' ');
+		$entry_data['abstract'] = preg_match("/Factory/",$entry_data['namespace']);
+		$entry_data['path'] = str_replace("/ILIAS","src",str_replace("\\","/",$entry_data['namespace']));
+
+		try{
+			$entry = new Entry\ComponentEntry($entry_data);
+		}catch(\Exception $e){
+			throw $this->ef->exception(Exception\CrawlerException::PARSING_YAML_ENTRY_FAILED,
+					" could not convert data to entry, message: '".$e->getMessage()."' file: ".$this->file_path);
+		}
+
+		return $entry;
+	}
```

Comment:
Will be done, while implementing the to visual representation of the KS. Since a rate this as hard to read, I will think about slightly refactoring it, to make it more comprehensible.